src/main/java/com/yelp/nrtsearch/server/handler/AddDocumentHandler.java

```diff
@@ -460,6 +473,58 @@ public long runIndexingJob() throws Exception {
       return shardState.writer.getMaxCompletedSequenceNumber();
     }
 
+    private void executeDocValueUpdateRequest(
+        DocumentsContext documentsContext,
+        IndexState indexState,
+        ShardState shardState,
+        AddDocumentRequest addDocumentRequest) {
+      try {
+        IndexingMetrics.updateDocValuesRequestsReceived.labelValues(indexName).inc();
+        List<Field> updatableDocValueFields = new ArrayList<>();
+        Term term = null;
+        for (Map.Entry<String, MultiValuedField> entry :
+            addDocumentRequest.getFieldsMap().entrySet()) {
+          FieldDef field = indexState.getField(entry.getKey());
+          if (field == null) {
+            throw new IllegalArgumentException(
+                String.format("Field: %s is not registered", entry.getKey()));
+          }
+          if (field.getName().equals(indexState.getIdFieldDef().get().getName())) {
+
+            String idFieldName = indexState.getIdFieldDef().get().getName();
+            String idFieldValue = addDocumentRequest.getFieldsMap().get(idFieldName).getValue(0);
+
+            if (idFieldValue == null || idFieldValue.isEmpty()) {
+              throw new IllegalArgumentException(
+                  String.format("the _ID should have a value set to execute update DocValue"));
+            }
+            term = new Term(idFieldName, idFieldValue);
+            continue;
+          }
```

Comment:
It might help with readability to create a separate method to extract the id term, then skip it when processing the other fields